src/js_results.hpp

```diff
@@ -254,6 +263,31 @@ void ResultsClass<T>::is_valid(ContextType ctx, ObjectType this_object, Argument
     return_value.set(get_internal<T, ResultsClass<T>>(this_object)->is_valid());
 }
 
+#if REALM_ENABLE_SYNC
+template<typename T>
+void ResultsClass<T>::subscribe(ContextType ctx, ObjectType this_object, Arguments args, ReturnValue &return_value) {
+    args.validate_maximum(1);
+
+    auto results = get_internal<T, ResultsClass<T>>(this_object);
+    auto realm = results->get_realm();
+    auto sync_config = realm->config().sync_config;
+    if (!sync_config || !sync_config->is_partial) {
+        throw std::logic_error("A partial sync query can only be registered in a partially synced Realm.");
```

Comment:
`partial_sync::subscribe` performs this check internally, so I don't think it's necessary to repeat it here?